### PR TITLE
Problem: Int64 is not checked in gas related api

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/backend/node_info.go [hunk 1]
```diff
@@ -345,10 +345,10 @@ func (b *Backend) RPCMinGasPrice() int64 {
...
 	minGasPrice := b.cfg.GetMinGasPrices()
-	amt := minGasPrice.AmountOf(evmParams.Params.EvmDenom).TruncateInt64()
-	if amt == 0 {
+	amt := minGasPrice.AmountOf(evmParams.Params.EvmDenom).TruncateInt()
+	if !amt.IsInt64() || amt.IsZero() {
 		return ethermint.DefaultGasPrice
 	}
 
-	return amt
+	return amt.Int64()
```
Comment (yihuang) [inline rpc/backend/node_info.go]: it should return `big.Int` in the first place?

Comment (mmsqe) [reply rpc/backend/node_info.go]: what if amt does exceeds MaxUint64

Comment (yihuang) [reply rpc/backend/node_info.go]: there should be no problem exceeding `MaxUInt64`, we do have 18 decimals.

File: rpc/backend/chain_info.go [hunk 2]
```diff
@@ -303,7 +303,7 @@ func (b *Backend) FeeHistory(
 // Although we don't support tx prioritization yet, but we return a positive value to help client to
 // mitigate the base fee changes.
 func (b *Backend) SuggestGasTipCap(baseFee *big.Int) (*big.Int, error) {
-	if baseFee == nil {
```
Comment (yihuang) [inline rpc/backend/chain_info.go]: this too, we should change the calculation below to use `big.Int`, similar to the `x/feemarket/keeper/eip1559.go`

Comment (mmsqe) [reply rpc/backend/chain_info.go]: actually it's impossible to have Int64() out of bound here

Comment (yihuang) [reply rpc/backend/chain_info.go]: the base fee won't exceed 10cro?

Comment (mmsqe) [reply rpc/backend/chain_info.go]: big.Int use Int64() is fine, math.Int is not

**Review by yihuang — APPROVED** [review]

File: rpc/backend/chain_info.go (hunk 2 above)
Comment (yihuang) [reply rpc/backend/chain_info.go]: it'll truncate silently? technically, that's not correct though.

**Pull request merged**